### PR TITLE
Add support for CDI in k8s-rdma-shared-dev-plugin deployments

### PR DESCRIPTION
**Pull request by almaslennikov**
Almost the same with **https://github.com/Mellanox/network-operator/pull/565**. Whichever is merged last needs to merge into the existing one

### CODE REVIEW

**Review by e0ne — CHANGES_REQUESTED** [review]
Please, update k8s-rdma-shared-dev-plugin version to the latest build `fe7f371c7e1b8315bf900f71cd25cfc1251dc775`

**Comment by abdallahyas** [comment]
CIs are unable to find the image, are you sure it exists?

File: deployment/network-operator/values.yaml [hunk 1]
```diff
@@ -198,8 +198,8 @@ ofedDriver:
 rdmaSharedDevicePlugin:
   deploy: true
   image: k8s-rdma-shared-dev-plugin
-  repository: nvcr.io/nvidia/cloud-native
-  version: v1.3.2
+  repository: ghcr.io/mellanox
+  version: fe7f371c7e1b8315bf900f71cd25cfc1251dc775
```
Comment (e0ne) [inline deployment/network-operator/values.yaml:202]: it should be `sha-fe7f371c7e1b8315bf900f71cd25cfc1251dc775`

**Review by e0ne — CHANGES_REQUESTED** [review]

**Review by e0ne — APPROVED** [review]

**Pull request merged**